cmd/wallet-web/src/pages/chapi/Signup.vue

```diff
@@ -0,0 +1,207 @@
+                     :src="provider.logoURL"/>
+                {{ provider.name }}
+              </a>
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import {DeviceLogin, RegisterWallet} from "./wallet"
+import {mapActions, mapGetters} from 'vuex'
+import {BeatLoader} from "@saeris/vue-spinners";
+import axios from 'axios';
+
+export default {
+  created: async function () {
+    await this.fetchProviders()
+    this.startLoading()
+    //TODO: issue-601 Implement cookie logic with information from the backend.
+    this.deviceLogin = new DeviceLogin(this.getAgentOpts());
+    let redirect = this.$route.params['redirect']
+    this.redirect = redirect ? {name: redirect} : `${__webpack_public_path__}`
+    this.loadUser()
+    if (this.getCurrentUser()) {
+      await this.getAgentInstance().store.flush()
+      this.handleSuccess()
+      return
+    }
+
+    await this.loadOIDCUser()
+
+    try {
+      await this.refreshUserMetadata()
+    } catch (e) {
+      console.warn("first time login: ignore", e)
+    }
+
+    if (this.getCurrentUser()) {
+      await this.finishOIDCLogin()
+      await this.getAgentInstance().store.flush()
+      this.handleSuccess()
+      return
+    }
+    if (this.$cookies.isKey('registerSuccess')) {
+      this.registered = true;
+    }
+
+    this.stopLoading()
+    this.loading = false
+  },
+  components: {
+    BeatLoader,
+  },
+  data() {
+    return {
+      providers:[],
+      hubOauthProvider: this.hubURL(),
+      statusMsg: '',
+      loading: true,
+      registered: false,
+      messageNum: 0,
+      messages: [
+        "Getting Started..",
+        "This could take a minute.",
+        "Please do not refresh the page.",
+        "Please wait...",
+      ]
+    };
+  },
+  methods: {
+    ...mapActions({
+      loadUser: 'loadUser',
+      loadOIDCUser: 'loadOIDCUser',
+      refreshUserMetadata: 'refreshUserMetadata',
+      startUserSetup: 'startUserSetup',
+      completeUserSetup: 'completeUserSetup'
+    }),
+    ...mapGetters(['getCurrentUser', 'getAgentOpts', 'serverURL', 'hubURL']),
+    ...mapGetters('agent', {getAgentInstance: 'getInstance'}),
+    beginOIDCLogin: async function () {
+      window.location.href = this.serverURL() + "/oidc/login"
+    },
+
+    // Fetching the providers from hub-auth
+    fetchProviders: async function() {
+      await axios.get(this.hubURL() +'/oauth2/providers')
+          .then(response => {
+            this.providers = response.data.authProviders
+          })
+    },
+
+    sleep(ms) {
+      return new Promise(resolve => setTimeout(resolve, ms));
+    },
+    finishOIDCLogin: async function () {
+      let user = this.getCurrentUser()
+
+      let registrar = new RegisterWallet(this.$polyfill, this.$webCredentialHandler, this.getAgentInstance(),
+          this.getAgentOpts())
+
+      try {
+        if (!user.metadata) {
```

Comment:
Could we also check if the user is there at first? Accessing properties from `undefined` throws errors.